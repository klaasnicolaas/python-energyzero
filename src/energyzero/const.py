"""Constants for EnergyZero API client."""
from __future__ import annotations

from enum import Enum


class VatOption(str, Enum):
    """Enum representing whether to include VAT or not."""

    INCLUDE = "true"
    EXCLUDE = "false"
