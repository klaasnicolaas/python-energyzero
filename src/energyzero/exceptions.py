"""Exceptions for EnergyZero."""


class EnergyZeroError(Exception):
    """Generic EnergyZero exception."""


class EnergyZeroConnectionError(EnergyZeroError):
    """EnergyZero - connection exception."""


class EnergyZeroNoDataError(EnergyZeroError):
    """EnergyZero - no data exception."""
