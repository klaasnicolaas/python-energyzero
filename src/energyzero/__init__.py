"""Asynchronous Python client for the EnergyZero API."""

from .energyzero import EnergyZero
from .exceptions import (
    EnergyZeroConnectionError,
    EnergyZeroError,
    EnergyZeroNoDataError,
)
from .models import Electricity, Gas

__all__ = [
    "Gas",
    "Electricity",
    "EnergyZero",
    "EnergyZeroError",
    "EnergyZeroNoDataError",
    "EnergyZeroConnectionError",
]
