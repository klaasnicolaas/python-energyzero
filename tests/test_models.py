"""Test the models."""
from datetime import date, datetime, timezone
from unittest.mock import Mock, patch

import aiohttp
import pytest
from aresponses import ResponsesMockServer

from energyzero import Electricity, EnergyZero, EnergyZeroNoDataError, Gas

from . import load_fixtures


@pytest.mark.asyncio
@patch(
    "energyzero.models.Electricity.utcnow",
    Mock(return_value=datetime(2022, 12, 7, 14, 0).replace(tzinfo=timezone.utc)),
)
async def test_electricity_model(aresponses: ResponsesMockServer) -> None:
    """Test the electricity model."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("energy.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        energy: Electricity = await client.energy_prices(
            start_date=today, end_date=today
        )
        assert energy is not None and isinstance(energy, Electricity)
        assert energy.extreme_prices[1] == 0.55
        assert energy.extreme_prices[0] == 0.26
        assert energy.average_price == 0.37
        assert energy.current_price == 0.48
        # The next hour price
        next_hour = datetime(2022, 12, 7, 15, 0).replace(tzinfo=timezone.utc)
        assert energy.price_at_time(next_hour) == 0.49
        assert energy.lowest_price_time == datetime.strptime(
            "2022-12-07 02:00", "%Y-%m-%d %H:%M"
        ).replace(tzinfo=timezone.utc)
        assert energy.highest_price_time == datetime.strptime(
            "2022-12-07 16:00", "%Y-%m-%d %H:%M"
        ).replace(tzinfo=timezone.utc)
        assert energy.pct_of_max_price == 87.27
        assert isinstance(energy.timestamp_prices, list)


@pytest.mark.asyncio
async def test_electricity_none_date(aresponses: ResponsesMockServer) -> None:
    """Test when there is no data for the current datetime."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("energy.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        energy: Electricity = await client.energy_prices(
            start_date=today, end_date=today
        )
        assert energy is not None and isinstance(energy, Electricity)
        assert energy.current_price is None
        assert energy.average_price == 0.37


@pytest.mark.asyncio
async def test_no_electricity_data(aresponses: ResponsesMockServer) -> None:
    """Raise exception when there is no data."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("no_data.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        with pytest.raises(EnergyZeroNoDataError):
            await client.energy_prices(start_date=today, end_date=today)


@pytest.mark.asyncio
@patch(
    "energyzero.models.Gas.utcnow",
    Mock(return_value=datetime(2022, 12, 7, 14, 0).replace(tzinfo=timezone.utc)),
)
async def test_gas_model(aresponses: ResponsesMockServer) -> None:
    """Test the gas model."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("gas.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        gas: Gas = await client.gas_prices(start_date=today, end_date=today)
        assert gas is not None and isinstance(gas, Gas)
        assert gas.extreme_prices[1] == 1.47
        assert gas.extreme_prices[0] == 1.43
        assert gas.average_price == 1.45
        assert gas.current_price == 1.47
        # The next hour price
        next_hour = datetime(2022, 12, 7, 15, 0).replace(tzinfo=timezone.utc)
        assert gas.price_at_time(next_hour) == 1.47


@pytest.mark.asyncio
@patch(
    "energyzero.energyzero.get_utcnow", Mock(return_value=datetime(2022, 12, 7, 4, 0))
)
@patch(
    "energyzero.models.Gas.utcnow",
    Mock(return_value=datetime(2022, 12, 7, 5, 0).replace(tzinfo=timezone.utc)),
)
async def test_gas_morning_model(aresponses: ResponsesMockServer) -> None:
    """Test the gas model in the morning."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("gas.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        gas: Gas = await client.gas_prices(start_date=today, end_date=today)
        assert gas is not None and isinstance(gas, Gas)
        assert gas.current_price == 1.47
        assert gas.average_price == 1.45


@pytest.mark.asyncio
async def test_gas_none_date(aresponses: ResponsesMockServer) -> None:
    """Test when there is no data for the current datetime."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("gas.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        gas: Gas = await client.gas_prices(start_date=today, end_date=today)
        assert gas is not None and isinstance(gas, Gas)
        assert gas.current_price is None
        assert gas.average_price == 1.45


@pytest.mark.asyncio
async def test_no_gas_data(aresponses: ResponsesMockServer) -> None:
    """Raise exception when there is no data."""
    aresponses.add(
        "api.energyzero.nl",
        "/v1/energyprices",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("no_data.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        today = date(2022, 12, 7)
        client = EnergyZero(session=session)
        with pytest.raises(EnergyZeroNoDataError):
            await client.gas_prices(start_date=today, end_date=today)
