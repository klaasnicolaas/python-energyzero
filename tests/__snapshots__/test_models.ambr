# serializer version: 1
# name: test_electricity_midnight_cest
  Electricity(prices={FakeDatetime(2022, 12, 6, 22, 0, tzinfo=datetime.timezone.utc): 0.31, FakeDatetime(2022, 12, 6, 23, 0, tzinfo=datetime.timezone.utc): 0.35, FakeDatetime(2022, 12, 7, 0, 0, tzinfo=datetime.timezone.utc): 0.32, FakeDatetime(2022, 12, 7, 1, 0, tzinfo=datetime.timezone.utc): 0.28, FakeDatetime(2022, 12, 7, 2, 0, tzinfo=datetime.timezone.utc): 0.26, FakeDatetime(2022, 12, 7, 3, 0, tzinfo=datetime.timezone.utc): 0.27, FakeDatetime(2022, 12, 7, 4, 0, tzinfo=datetime.timezone.utc): 0.28, FakeDatetime(2022, 12, 7, 5, 0, tzinfo=datetime.timezone.utc): 0.28, FakeDatetime(2022, 12, 7, 6, 0, tzinfo=datetime.timezone.utc): 0.38, FakeDatetime(2022, 12, 7, 7, 0, tzinfo=datetime.timezone.utc): 0.41, FakeDatetime(2022, 12, 7, 8, 0, tzinfo=datetime.timezone.utc): 0.46, FakeDatetime(2022, 12, 7, 9, 0, tzinfo=datetime.timezone.utc): 0.44, FakeDatetime(2022, 12, 7, 10, 0, tzinfo=datetime.timezone.utc): 0.39, FakeDatetime(2022, 12, 7, 11, 0, tzinfo=datetime.timezone.utc): 0.33, FakeDatetime(2022, 12, 7, 12, 0, tzinfo=datetime.timezone.utc): 0.37, FakeDatetime(2022, 12, 7, 13, 0, tzinfo=datetime.timezone.utc): 0.44, FakeDatetime(2022, 12, 7, 14, 0, tzinfo=datetime.timezone.utc): 0.48, FakeDatetime(2022, 12, 7, 15, 0, tzinfo=datetime.timezone.utc): 0.49, FakeDatetime(2022, 12, 7, 16, 0, tzinfo=datetime.timezone.utc): 0.55, FakeDatetime(2022, 12, 7, 17, 0, tzinfo=datetime.timezone.utc): 0.37, FakeDatetime(2022, 12, 7, 18, 0, tzinfo=datetime.timezone.utc): 0.4, FakeDatetime(2022, 12, 7, 19, 0, tzinfo=datetime.timezone.utc): 0.4, FakeDatetime(2022, 12, 7, 20, 0, tzinfo=datetime.timezone.utc): 0.32, FakeDatetime(2022, 12, 7, 21, 0, tzinfo=datetime.timezone.utc): 0.33, FakeDatetime(2022, 12, 7, 22, 0, tzinfo=datetime.timezone.utc): 0.31}, average_price=0.37)
# ---
# name: test_electricity_model
  Electricity(prices={FakeDatetime(2022, 12, 6, 22, 0, tzinfo=datetime.timezone.utc): 0.31, FakeDatetime(2022, 12, 6, 23, 0, tzinfo=datetime.timezone.utc): 0.35, FakeDatetime(2022, 12, 7, 0, 0, tzinfo=datetime.timezone.utc): 0.32, FakeDatetime(2022, 12, 7, 1, 0, tzinfo=datetime.timezone.utc): 0.28, FakeDatetime(2022, 12, 7, 2, 0, tzinfo=datetime.timezone.utc): 0.26, FakeDatetime(2022, 12, 7, 3, 0, tzinfo=datetime.timezone.utc): 0.27, FakeDatetime(2022, 12, 7, 4, 0, tzinfo=datetime.timezone.utc): 0.28, FakeDatetime(2022, 12, 7, 5, 0, tzinfo=datetime.timezone.utc): 0.28, FakeDatetime(2022, 12, 7, 6, 0, tzinfo=datetime.timezone.utc): 0.38, FakeDatetime(2022, 12, 7, 7, 0, tzinfo=datetime.timezone.utc): 0.41, FakeDatetime(2022, 12, 7, 8, 0, tzinfo=datetime.timezone.utc): 0.46, FakeDatetime(2022, 12, 7, 9, 0, tzinfo=datetime.timezone.utc): 0.44, FakeDatetime(2022, 12, 7, 10, 0, tzinfo=datetime.timezone.utc): 0.39, FakeDatetime(2022, 12, 7, 11, 0, tzinfo=datetime.timezone.utc): 0.33, FakeDatetime(2022, 12, 7, 12, 0, tzinfo=datetime.timezone.utc): 0.37, FakeDatetime(2022, 12, 7, 13, 0, tzinfo=datetime.timezone.utc): 0.44, FakeDatetime(2022, 12, 7, 14, 0, tzinfo=datetime.timezone.utc): 0.48, FakeDatetime(2022, 12, 7, 15, 0, tzinfo=datetime.timezone.utc): 0.49, FakeDatetime(2022, 12, 7, 16, 0, tzinfo=datetime.timezone.utc): 0.55, FakeDatetime(2022, 12, 7, 17, 0, tzinfo=datetime.timezone.utc): 0.37, FakeDatetime(2022, 12, 7, 18, 0, tzinfo=datetime.timezone.utc): 0.4, FakeDatetime(2022, 12, 7, 19, 0, tzinfo=datetime.timezone.utc): 0.4, FakeDatetime(2022, 12, 7, 20, 0, tzinfo=datetime.timezone.utc): 0.32, FakeDatetime(2022, 12, 7, 21, 0, tzinfo=datetime.timezone.utc): 0.33, FakeDatetime(2022, 12, 7, 22, 0, tzinfo=datetime.timezone.utc): 0.31}, average_price=0.37)
# ---
# name: test_electricity_none_date
  Electricity(prices={datetime.datetime(2022, 12, 6, 22, 0, tzinfo=datetime.timezone.utc): 0.31, datetime.datetime(2022, 12, 6, 23, 0, tzinfo=datetime.timezone.utc): 0.35, datetime.datetime(2022, 12, 7, 0, 0, tzinfo=datetime.timezone.utc): 0.32, datetime.datetime(2022, 12, 7, 1, 0, tzinfo=datetime.timezone.utc): 0.28, datetime.datetime(2022, 12, 7, 2, 0, tzinfo=datetime.timezone.utc): 0.26, datetime.datetime(2022, 12, 7, 3, 0, tzinfo=datetime.timezone.utc): 0.27, datetime.datetime(2022, 12, 7, 4, 0, tzinfo=datetime.timezone.utc): 0.28, datetime.datetime(2022, 12, 7, 5, 0, tzinfo=datetime.timezone.utc): 0.28, datetime.datetime(2022, 12, 7, 6, 0, tzinfo=datetime.timezone.utc): 0.38, datetime.datetime(2022, 12, 7, 7, 0, tzinfo=datetime.timezone.utc): 0.41, datetime.datetime(2022, 12, 7, 8, 0, tzinfo=datetime.timezone.utc): 0.46, datetime.datetime(2022, 12, 7, 9, 0, tzinfo=datetime.timezone.utc): 0.44, datetime.datetime(2022, 12, 7, 10, 0, tzinfo=datetime.timezone.utc): 0.39, datetime.datetime(2022, 12, 7, 11, 0, tzinfo=datetime.timezone.utc): 0.33, datetime.datetime(2022, 12, 7, 12, 0, tzinfo=datetime.timezone.utc): 0.37, datetime.datetime(2022, 12, 7, 13, 0, tzinfo=datetime.timezone.utc): 0.44, datetime.datetime(2022, 12, 7, 14, 0, tzinfo=datetime.timezone.utc): 0.48, datetime.datetime(2022, 12, 7, 15, 0, tzinfo=datetime.timezone.utc): 0.49, datetime.datetime(2022, 12, 7, 16, 0, tzinfo=datetime.timezone.utc): 0.55, datetime.datetime(2022, 12, 7, 17, 0, tzinfo=datetime.timezone.utc): 0.37, datetime.datetime(2022, 12, 7, 18, 0, tzinfo=datetime.timezone.utc): 0.4, datetime.datetime(2022, 12, 7, 19, 0, tzinfo=datetime.timezone.utc): 0.4, datetime.datetime(2022, 12, 7, 20, 0, tzinfo=datetime.timezone.utc): 0.32, datetime.datetime(2022, 12, 7, 21, 0, tzinfo=datetime.timezone.utc): 0.33, datetime.datetime(2022, 12, 7, 22, 0, tzinfo=datetime.timezone.utc): 0.31}, average_price=0.37)
# ---
# name: test_gas_model
  Gas(prices={FakeDatetime(2022, 12, 5, 23, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 0, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 1, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 2, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 3, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 4, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 5, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 6, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 7, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 8, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 9, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 10, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 11, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 12, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 13, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 14, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 15, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 16, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 17, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 18, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 19, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 20, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 21, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 22, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 23, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 0, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 1, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 2, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 3, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 4, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 5, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 6, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 7, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 8, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 9, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 10, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 11, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 12, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 13, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 14, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 15, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 16, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 17, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 18, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 19, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 20, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 21, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 22, 0, tzinfo=datetime.timezone.utc): 1.47}, average_price=1.46)
# ---
# name: test_gas_morning_model
  Gas(prices={FakeDatetime(2022, 12, 5, 23, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 0, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 1, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 2, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 3, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 4, 0, tzinfo=datetime.timezone.utc): 1.43, FakeDatetime(2022, 12, 6, 5, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 6, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 7, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 8, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 9, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 10, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 11, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 12, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 13, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 14, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 15, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 16, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 17, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 18, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 19, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 20, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 21, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 22, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 6, 23, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 0, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 1, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 2, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 3, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 4, 0, tzinfo=datetime.timezone.utc): 1.45, FakeDatetime(2022, 12, 7, 5, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 6, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 7, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 8, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 9, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 10, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 11, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 12, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 13, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 14, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 15, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 16, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 17, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 18, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 19, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 20, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 21, 0, tzinfo=datetime.timezone.utc): 1.47, FakeDatetime(2022, 12, 7, 22, 0, tzinfo=datetime.timezone.utc): 1.47}, average_price=1.46)
# ---
# name: test_gas_none_date
  Gas(prices={datetime.datetime(2022, 12, 5, 23, 0, tzinfo=datetime.timezone.utc): 1.43, datetime.datetime(2022, 12, 6, 0, 0, tzinfo=datetime.timezone.utc): 1.43, datetime.datetime(2022, 12, 6, 1, 0, tzinfo=datetime.timezone.utc): 1.43, datetime.datetime(2022, 12, 6, 2, 0, tzinfo=datetime.timezone.utc): 1.43, datetime.datetime(2022, 12, 6, 3, 0, tzinfo=datetime.timezone.utc): 1.43, datetime.datetime(2022, 12, 6, 4, 0, tzinfo=datetime.timezone.utc): 1.43, datetime.datetime(2022, 12, 6, 5, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 6, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 7, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 8, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 9, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 10, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 11, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 12, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 13, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 14, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 15, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 16, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 17, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 18, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 19, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 20, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 21, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 22, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 6, 23, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 7, 0, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 7, 1, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 7, 2, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 7, 3, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 7, 4, 0, tzinfo=datetime.timezone.utc): 1.45, datetime.datetime(2022, 12, 7, 5, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 6, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 7, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 8, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 9, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 10, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 11, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 12, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 13, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 14, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 15, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 16, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 17, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 18, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 19, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 20, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 21, 0, tzinfo=datetime.timezone.utc): 1.47, datetime.datetime(2022, 12, 7, 22, 0, tzinfo=datetime.timezone.utc): 1.47}, average_price=1.46)
# ---
